### PR TITLE
feat(pruner, metrics): `skip` attribute for metrics derive macro

### PR DESCRIPTION
**Pull request by shekhirin**
Allows to skip certain struct fields which are not intended to be parsed as metrics: https://github.com/paradigmxyz/reth/blob/bf37faf586fe4538b5b7e675f5ab31ba93c14dc8/crates/prune/src/metrics.rs#L5-L12

### CODE REVIEW

File: crates/metrics/metrics-derive/src/expand.rs [hunk 1]
```diff
@@ -246,40 +301,57 @@ fn parse_metrics_attr(node: &DeriveInput) -> Result<MetricsAttr> {
...
-                        return Err(Error::new_spanned(kv, "Duplicate `rename` value provided."))
+            let parsed =
+                metric_attr.parse_args_with(Punctuated::<Meta, Token![,]>::parse_terminated)?;
+            for meta in parsed {
+                match meta {
+                    Meta::Path(path) if path.is_ident("skip") => skip = true,
+                    Meta::NameValue(kv) => {
+                        let lit = match kv.value {
+                            Expr::Lit(ref expr) => &expr.lit,
+                            _ => continue,
```
Comment (mattsse) [inline crates/metrics/metrics-derive/src/expand.rs:321]: simple

**Review by mattsse — APPROVED** [review]
reasonable feature.

lgtm

File: crates/metrics/metrics-derive/src/expand.rs [hunk 2]
```diff
@@ -93,37 +129,56 @@ pub(crate) fn derive(node: &DeriveInput) -> Result<proc_macro2::TokenStream> {
...
-                .fold((vec![], vec![], vec![]), |mut acc, x| {
-                    acc.0.push(x.0);
-                    acc.1.push(x.1);
-                    acc.2.push(x.2);
-                    acc
-                });
+                .fold(
+                    (vec![], vec![], vec![]),
+                    |(mut defaults, mut labeled_defaults, mut describes),
+                     (default, labeled_default, describe)| {
```
Comment (rkrasiuk) [inline crates/metrics/metrics-derive/src/expand.rs:173]: let's avoid using keywords as var names

**Review by rkrasiuk — CHANGES_REQUESTED** [review]
one smol nit. could we also get tests the skip fields being at the end of the struct, in the middle and having multiple consecutive/separate skip fields

File: crates/metrics/metrics-derive/src/expand.rs (hunk 2 above)
Comment (shekhirin) [reply crates/metrics/metrics-derive/src/expand.rs]: ah, didn't know it's a keyword from nightly feature, fixed

**Review by mattsse — APPROVED** [review]

**Review by rkrasiuk — APPROVED** [review]
beautiful

**Pull request merged**